### ISSUE TITLE
writing mathML: TypeError (not iterable)

### ISSUE THREAD

**Issue by Karrenbelt** [sympy/sympy#17183]
```
import sympy
x, y = sympy.symbols('x y')
sympy.__version__
>>> '1.3'
```

I'm running into mathml printing issues trying to write some sympy constants / logic

Writing the first two constants works, last two fail
```
sympy.printing.mathml(sympy.pi)
sympy.printing.mathml(sympy.E)
sympy.printing.mathml(sympy.true) # 
sympy.printing.mathml(sympy.false) # 
```

```
>>> '<pi/>'
>>> '<exponentiale/>'
>>> TypeError: 'BooleanTrue' object is not iterable
>>> TypeError: 'BooleanTrue' object is not iterable
```

Writing the first two logical functions works, last two fail

```
sympy.printing.mathml(sympy.And(x,y))
sympy.printing.mathml(sympy.Or(x,y)) 
sympy.printing.mathml(sympy.Not(x))
sympy.printing.mathml(sympy.Xor(x,y))
```

```
>>> '<apply><and/><ci>x</ci><ci>y</ci></apply>'
>>> '<apply><or/><ci>x</ci><ci>y</ci></apply>'
>>> TypeError: 'Not' object is not iterable
>>> TypeError: 'Xor' object is not iterable
```

**Comment by Karrenbelt** [comment]
I also obtained this error with trying to convert lambda
```
sympy.printing.mathml(sympy.Lambda(y, y**y))
>>> TypeError: 'Lambda' object is not iterable
```

**Comment by oscargus** [comment]
These were fixed a few days ago in  #17194 (although I didn't see this issue at the time it seems to have been spot on...).
